scripts/protocgen.sh

```diff
@@ -7,36 +7,27 @@
 set -eo pipefail
 
 protoc_gen_gocosmos() {
-  if ! grep "github.com/gogo/protobuf => github.com/regen-network/protobuf" go.mod &>/dev/null ; then
+  if ! grep "github.com/cosmos/gogoproto" go.mod &>/dev/null ; then
     echo -e "\tPlease run this command from somewhere inside the cosmos-sdk folder."
     return 1
   fi
 
-  go get github.com/regen-network/cosmos-proto/protoc-gen-gocosmos@latest 2>/dev/null
+  go get github.com/cosmos/gogoproto/protoc-gen-gocosmos@latest
 }
 
 protoc_gen_gocosmos
 
 proto_dirs=$(find ./proto -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq)
 for dir in $proto_dirs; do
-  buf protoc \
-    -I "proto" \
-    -I "third_party/proto" \
-    --gocosmos_out=plugins=grpc,\
-Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
-    --grpc-gateway_out=logtostderr=true,allow_colon_final_segments=true:. \
-  $(find "${dir}" -maxdepth 1 -name '*.proto')
-
+  proto_files=$(find "${dir}" -maxdepth 1 -name '*.proto')
+  for file in $proto_files; do
+    # Check if the go_package in the file is pointing to provenance
+    if grep -q "option go_package.*provenance" "$file"; then
+      buf generate --template proto/buf.gen.gogo.yaml "$file"
+    fi
+  done
 done
 
-# command to generate docs using protoc-gen-doc
-buf protoc \
-  -I "proto" \
-  -I "third_party/proto" \
-  --doc_out=./docs \
-  --doc_opt=./docs/protodoc-markdown.tmpl,proto-docs.md \
-  $(find "$(pwd)/proto" -maxdepth 5 -name '*.proto')
-
```

Comment:
We should probably figure out what (if anything) consumes that `proto-docs.md` file so we can know if its safe to not generate anymore. In my past explorations with the updated `buf` stuff, I couldn't find a straight-forward replacement for this, though. So if anything still needs that content, we might have to come up with a different solution for it.

Reply:
I pinged AJ on this. He said he doesn't know of anything that uses it.

So we're probably okay not generating it anymore, and should probably also delete the `docs/proto-docs.md` file (the thing this generated).